connector.go

```diff
@@ -104,6 +107,63 @@ type Connector struct {
 	requestor      HttpRequestor
 }
 
+func getLatestVersion(hostCfg HostConfig, authCfg AuthConfig, transportCfg TransportConfig) (string, error) {
+	schemaUrl := fmt.Sprintf("https://%s/wapi/v1.0/?_schema", hostCfg.Host)
+	req, err := http.NewRequest(http.MethodGet, schemaUrl, nil)
+	if err != nil {
+		return "", err
+	}
+
+	req.SetBasicAuth(authCfg.Username, authCfg.Password)
```

Comment:
What will happen in the case when we do not use user's name and password at all? (certificate-based client's authentication)

Reply:
Updated the logic. Now client will use the requestor object, which uses client certs, and if username/password are configured, I will add them to the request.